File: fixtures/command-injection-go/vuln.go
package main

import (
  "os/exec"
)

// PLANTED VULNERABILITY (do not "fix"): the untrusted `payload` argument is forwarded as the shell
// command tail, so payloads like `; echo RAEUBER_x` are executed by `/bin/sh -c`.
func run(payload string) string {
  out, err := exec.Command("sh", "-c", payload).CombinedOutput()
  if err != nil {
    return err.Error()
  }
  return string(out)
}

File: fixtures/command-injection-python/vuln.py
import subprocess

# PLANTED VULNERABILITY (do not "fix"): the untrusted `cmd` argument is executed through
# `subprocess.run(..., shell=True)`, so payloads like `; echo RAEUBER_x` are directly passed to
# `/bin/sh`. This fixture must be exploitable so the lane canary remains LIVE (or it is quarantined).
def run(cmd):
  return subprocess.run(cmd, shell=True, check=False, capture_output=True, text=True).stdout
